modules/apps/dynamic-data-lists/dynamic-data-lists-web/src/main/resources/META-INF/resources/configuration.jsp

```diff
@@ -26,9 +26,19 @@ boolean editable = PrefsParamUtil.getBoolean(PortletPreferencesFactoryUtil.getPo
 boolean formView = PrefsParamUtil.getBoolean(PortletPreferencesFactoryUtil.getPortletSetup(renderRequest), renderRequest, "formView", false);
 long formDDMTemplateId = PrefsParamUtil.getLong(PortletPreferencesFactoryUtil.getPortletSetup(renderRequest), renderRequest, "formDDMTemplateId");
 long recordSetId = PrefsParamUtil.getLong(PortletPreferencesFactoryUtil.getPortletSetup(renderRequest), renderRequest, "recordSetId");
+String scopeType = PrefsParamUtil.getString(PortletPreferencesFactoryUtil.getPortletSetup(renderRequest), renderRequest, "lfrScopeType");
 boolean spreadsheet = PrefsParamUtil.getBoolean(PortletPreferencesFactoryUtil.getPortletSetup(renderRequest), renderRequest, "spreadsheet");
 
-Group scopeGroup = themeDisplay.getScopeGroup();
+Group scopeGroup = null;
+
+if (scopeType.equals("company")) {
+	scopeGroup = GroupLocalServiceUtil.getGroup(themeDisplay.getCompanyGroupId());
+
+	scopeGroupId = scopeGroup.getGroupId();
+}
+else {
+	scopeGroup = themeDisplay.getScopeGroup();
+}
 
 if (scopeGroup.isStagingGroup() && !scopeGroup.isInStagingPortlet(DDLPortletKeys.DYNAMIC_DATA_LISTS)) {
 	scopeGroupId = scopeGroup.getLiveGroupId();
```

Comment:
Hi @IstvanD, that's not exactly what I meant.. I was talking about doing something like:

```
Group scopeGroup = themeDisplay.getScopeGroup();

if (scopeType.equals("company")) {
	scopeGroup = GroupLocalServiceUtil.getGroup(themeDisplay.getCompanyGroupId());

	scopeGroupId = scopeGroup.getGroupId();
}
else if (scopeGroup.isStagingGroup() && !scopeGroup.isInStagingPortlet(DDLPortletKeys.DYNAMIC_DATA_LISTS)) {
    scopeGroupId = scopeGroup.getLiveGroupId();
}
```

Reply:
Because this second `if` is also changing the `scopeGroupId` value.

Reply:
@carolmariaabb I see. The scopeGroup can be a staging group even when it is a company group. The Global site can also be staged. Therefore, they have to be checked in a distinct clause.

Reply:
Got it, @IstvanD. In this case, you can go back to the previous code, without the `if/else`. Thanks!

Reply:
Hi @carolmariaabb I have reverted the changes.